ushahidi/platform-client
# 3793 description feature

This pull request makes the following changes:
- Adds markdown-editor to field-attribute description
- Adds scss to resolve conflicts with pattern-library css
- Displays description in markdown in post-edit/post-add views
- Removes toggle-switch for description and default-values

Testing checklist:
- Go to settings-survey
- Add a new survey
- Add a new field, or edit title or description
- [x] A markdown editor should appear below "Add field description (optional:)":
![Screenshot 2019-11-11 at 10 43 38](https://user-images.githubusercontent.com/8624777/68577365-39c0b400-0470-11ea-82f7-08e0486c09ee.png)
- [ ] For fields where it's possible to add a default value  (all field-types except "upload", "description", "title" and "tags"), the default value box should be displayed with no toggle-switch:
![Screenshot 2019-11-11 at 10 46 28](https://user-images.githubusercontent.com/8624777/68577529-91f7b600-0470-11ea-9bbd-2ed95665a362.png)

- Add some text in the description-editor, format it however you want to, and as long text as you want to
- Save the field
- Save the survey
- Choose to add a new post to this survey
- [x] The description should be visible for the field you added description
- [x] The description should have the same formatting as you did in the editor
- Save the post
- Go to data-view
- Select the post you added and select "edit"
- [x] The description should be visible for the field you added description
- [x] The description should have the same formatting as you did in the editor

Repeat the above steps for a survey that already exists

- [ x] I certify that I ran my checklist

Fixes ushahidi/platform#3793 .

Ping @ushahidi/platform
